### ISSUE TITLE
get the query result in the main body ?

### ISSUE THREAD

**Issue by wizarden** [hgourvest/node-firebird#152]
Firebird.attach(options, function(err, db) { 
  if (err) throw err; 
db.query('select * from tab', function(err, result) {
    db.detach();
    console.log(result) // ok
  });
});

console.log(result) // undefined

**Comment by slachtar** [comment]
IMHO It does make sens since the 2ND call to console.log(result) is called
early before the result is defined which happens later when query returns
result.

**Comment by ericlovescode** [comment]
Check whether err is defined and what it contains.

So in your query function you will have:

  db.detach();
  if (err) {
    console.log(err);
  } else {
    console.log(result);
  }

**Comment by wizarden** [comment]
you need to do everything inside db.query

**Comment by wizarden** [comment]
you need to do everything inside db.query
